services/app-api/resolvers/indexSubmissions2.test.ts

```diff
@@ -0,0 +1,200 @@
+                },
+            },
+        })
+
+        // First, create new submissions
+        const draftSubmission = await createTestDraftSubmission(server)
+        const submittedSubmission = await createTestStateSubmission(server)
+        const unlockedSubmission = await createTestStateSubmission(server)
+        const relockedSubmission = await createTestStateSubmission(server)
+
+        // unlock two
+        await unlockTestDraftSubmission(cmsServer, unlockedSubmission.id)
+        await unlockTestDraftSubmission(cmsServer, relockedSubmission.id)
+
+        // resubmit one
+        await submitTestDraftSubmission(server, relockedSubmission.id)
+
+        // index submissions api request
+        const result = await server.executeOperation({
+            query: INDEX_SUBMISSIONS2,
+        })
+        const submissionsIndex = result.data?.indexSubmissions2
+        
+        // pull out test related submissions and order them
+        const testSubmissionIDs = [
+            draftSubmission.id,
+            submittedSubmission.id,
+            unlockedSubmission.id,
+            relockedSubmission.id,
+        ]
+        const testSubmissions: Submission2[] = submissionsIndex.edges
+            .map((edge: Submission2Edge) => edge.node)
+            .filter((test: Submission2) => testSubmissionIDs.includes(test.id))
+        
+        expect(testSubmissions.length).toBe(4)
+            
+        // organize test submissions in a predictable order via testSubmissionsIds array 
+        testSubmissions.sort((a, b) => {
+            if (
+                testSubmissionIDs.indexOf(a.id) >
+                testSubmissionIDs.indexOf(b.id)
+            ) {
+                return 1
+            } else {
+                return -1
+            }
+        })
+
+        expect(testSubmissions[0].status).toBe('DRAFT')
+        expect(testSubmissions[0].status).toBe('DRAFT')
+        expect(testSubmissions[0].status).toBe('DRAFT')
+        expect(testSubmissions[0].status).toBe('DRAFT')
+    })
+
+
+    it('a different user from the same state can index submissions', async () => {
+         const server = await constructTestPostgresServer()
+
+         // First, create a new submission
+         const stateSubmission = await createTestStateSubmission(server)
+
+         const createdID = stateSubmission.id
+
+         // then see if we can fetch that same submission
+         const input = {
+             submissionID: createdID,
+         }
+
+         // setup a server with a different user
+         const otherUserServer = await constructTestPostgresServer({
+             context: {
+                 user: {
+                     name: 'Aang',
+                     state_code: 'FL',
+                     role: 'STATE_USER',
+                     email: 'aang@mn.gov',
+                 },
+             },
+         })
+
+         // TODO should we type these results
+         const result = await otherUserServer.executeOperation({
```

Comment:
q for @macrael  - I noticed (or maybe re-discovered) that `result` isn't typed. I was trying to figure out a way to have it infer the type of the GraphQLReponse for a specific query but I couldn't. Any ideas here?

Reply:
I’m not sure. So, this type is the return type we get from the client code, which we generate in app-web. Maybe we could generate those types for app-api and set the types right